nautobot/core/tests/integration/test_navbar.py

```diff
@@ -0,0 +1,175 @@
+        user_permissions = self.user.get_all_permissions()
+
+        self.selenium.get(f"{self.live_server_url}")
+        self.selenium.wait_for_html("body")
+
+        for tab_name, groups in self.navbar.items():
+            # XPath to find tabs using the tab name
+            tab_xpath = f"//*[@id='navbar']//*[contains(text(), '{tab_name}')]"
+            tab = self.selenium.find_element_by_xpath(tab_xpath)
+            tab.click()
+            self.assertTrue(bool(tab.get_attribute("aria-expanded")))
+
+            for group_name, items in groups.items():
+                # Append onto tab xpath with group name search
+                group = tab.find_element_by_xpath(
+                    f"{tab_xpath}/following-sibling::ul//li[contains(text(), '{group_name}')]"
+                )
+
+                for item_name, item_details in items.items():
+                    item_xpath = f"{tab_xpath}/following-sibling::ul//li[.//a[contains(text(), '{item_name}')]]"
+                    item = group.find_element_by_xpath(item_xpath)
+                    if item_details["permission"] in user_permissions:
+                        self.assertNotEquals(
+                            item.get_attribute("class"), "disabled", f"Item `{item_name}` should not be disabled."
+                        )
+                    else:
+                        self.assertEquals(
+                            item.get_attribute("class"), "disabled", f"Item `{item_name}` should be disabled."
+                        )
+
+    @override_settings(HIDE_RESTRICTED_UI=False)
+    def test_navbar_render_no_permissions(self):
+        """
+        Render navbar from home page with no permissions.
+        """
+        self.selenium.get(f"{self.live_server_url}")
+        self.selenium.wait_for_html("body")
+
+        for tab_name, groups in self.navbar.items():
+            # XPath to find tabs using the tab name
+            tab_xpath = f"//*[@id='navbar']//*[contains(text(), '{tab_name}')]"
+            tab = self.selenium.find_element_by_xpath(tab_xpath)
+            tab.click()
+            self.assertTrue(bool(tab.get_attribute("aria-expanded")))
+
+            for group_name, items in groups.items():
+                # Append onto tab xpath with group name search
+                group = tab.find_element_by_xpath(
+                    f"{tab_xpath}/following-sibling::ul//li[contains(text(), '{group_name}')]"
+                )
+
+                for item_name, _ in items.items():
+                    item_xpath = f"{tab_xpath}/following-sibling::ul//li[.//a[contains(text(), '{item_name}')]]"
+                    item = group.find_element_by_xpath(item_xpath)
+                    self.assertEquals(
+                        item.get_attribute("class"), "disabled", f"Item `{item_name}` should be disabled."
+                    )
+
+    @override_settings(HIDE_RESTRICTED_UI=True)
+    def test_navbar_render_restricted_ui(self):
```

Comment:
I'd prefer if this test were more similar to `test_navbar_render_limit_permissions` rather than `test_navbar_render_no_permissions` -- in other words, it's a more interesting test if we have a mixture of "should render" and "should not render" menu items, rather than this case where all menu items should not be rendered since the user has no permissions at all.